sahi/model.py

```diff
@@ -462,16 +460,21 @@ def _create_object_prediction_list_from_original_predictions(
                 category_id = int(prediction[5])
                 category_name = self.category_mapping[str(category_id)]
 
+                # fix negative box coords
+                bbox[0] = max(0, bbox[0])
+                bbox[1] = max(0, bbox[1])
+                bbox[2] = max(0, bbox[2])
+                bbox[3] = max(0, bbox[3])
+
+                # fix out of image box coords
+                if full_shape is not None:
+                    bbox[0] = min(full_shape[1], bbox[0])
+                    bbox[1] = min(full_shape[0], bbox[1])
+                    bbox[2] = min(full_shape[1], bbox[2])
+                    bbox[3] = min(full_shape[0], bbox[3])
+
                 # ignore invalid predictions
-                if bbox[0] > bbox[2] or bbox[1] > bbox[3] or bbox[0] < 0 or bbox[1] < 0 or bbox[2] < 0 or bbox[3] < 0:
-                    logger.warning(f"ignoring invalid prediction with bbox: {bbox}")
-                    continue
-                if full_shape is not None and (
-                    bbox[1] > full_shape[0]
-                    or bbox[3] > full_shape[0]
-                    or bbox[0] > full_shape[1]
-                    or bbox[2] > full_shape[1]
-                ):
+                if not (bbox[0] < bbox[2]) or not (bbox[1] < bbox[3]):
```

Comment:
Aynı kod bloğu 322-336'da da var. Duplication'ı önlemek için helper function yazılabilir.